app/scripts/services/authorization.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("AuthorizationService", function($q, Logger, $interval, DataService, APIService){
+    function AuthorizationService() {
+      this.rules = null;
+      this.intervalTimer = null;
+      this.forceReload = false;
+      this.canICreateSomeResource = false;
+      this.projectNameRules = null;
+    }
+
+    var normalizeRules = function(rules) {
```

Comment:
I know for sure we need to be handling getting back a "*" for a resource or a "*" for the verb.  I think a resource here might also be able to be a specific resource including the name.  @liggitt can you point us to the API doc for what all the possibilities are for these rules